src/js/pages/Index.js

```diff
@@ -41,6 +42,8 @@ var Index = React.createClass({
     MetadataStore.init();
     SidebarStore.init();
 
+    NavigationServiceUtil.registerRoutesInNavigation(this.props.routes);
```

Comment:
After discussion, with @nLight we agreed that this utility is more like a "Legacy Glue" and therefore kind of makes sense to be separated. 👍